src/Components/Modal/edit-profile-modal.jsx

```diff
@@ -0,0 +1,85 @@
+import { useState } from 'react';
+import { BsX } from 'react-icons/bs';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchEditUser } from '../../features/users/userSlice';
+
+export function EditProfile({ setShowModal, setImg, img }) {
+
+  const dispatch = useDispatch();
+  const userProfileData = useSelector(state => state.userProfile);
+  const authUserData = useSelector(state => state.user);
+  const [userData, setUserData] = useState({...authUserData.user});
+  const token = authUserData.token;
+
+
+  const userDataChangeHandler = (e) => {
+    setUserData(pre => ({...pre, [e.target.name] : e.target.value}));
+  }
+
+  const updateDataHandler = () => {
+    dispatch(fetchEditUser({userData, token}));
+    setShowModal(false);
+  }
+
+  const imgHandler = (e) => {
+    let reader = new FileReader();
+
+    reader.readAsDataURL(e.target.files[0]);
+
+    reader.onload = () => {
+      if (reader.readyState === 2) {
+        setImg(reader.result);
+      }
+    }
+  }
+
+  return (
+    <>
+          <div
+            className="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
+          >
+            <div className="relative w-full my-6 mx-auto max-w-[24rem]">
+             
+              <div className="border-4 border-black-300 p-4 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none">     
+                 <div className="w-full text-[2.6rem] cursor-pointer flex justify-end"><BsX onClick={() => setShowModal(false)} /></div>
+                 <div className="flex flex-col gap-6 ">
+                        <div className="flex items-center">
+                            <label htmlFor="avatar">Avatar</label>
+                            <div className=" ml-[5rem]">
+                                <label className="cursor-pointer">
+                                  <img className='w-14 h-14 rounded-full' src={img} alt="profile-img" />
+                                  <input onChange={(e) => imgHandler(e)} accept="image/*" type="file" className="hidden" />
```

Comment:
you do not need to pass `e` from here in imgHandler. You still have access to the event object where you have defined the function